### ISSUE TITLE
open idea 2019.3 EAP error

### ISSUE THREAD

**Issue by afsfge** [JetBrains/idea-gitignore#606]
### Prerequisites

* [X] Plugin is in the latest version
* [X] Issue was not reported yet
* [X] Stack trace (if provided) contains `mobi.hsz.idea.gitignore` package name

### Description

```
Plugin 'mobi.hsz.idea.gitignore' failed to initialize and will be disabled.  Please restart IntelliJ IDEA.

java.lang.IllegalAccessError: failed to access class com.intellij.util.containers.ConcurrentWeakHashMap from class mobi.hsz.idea.gitignore.util.CachedConcurrentMap (com.intellij.util.containers.ConcurrentWeakHashMap is in unnamed module of loader com.intellij.util.lang.UrlClassLoader @c038203; mobi.hsz.idea.gitignore.util.CachedConcurrentMap is in unnamed module of loader com.intellij.ide.plugins.cl.PluginClassLoader @7ded9ef3)
    at mobi.hsz.idea.gitignore.util.CachedConcurrentMap.<init>(CachedConcurrentMap.java:53)
    at mobi.hsz.idea.gitignore.util.CachedConcurrentMap.create(CachedConcurrentMap.java:65)
    at mobi.hsz.idea.gitignore.IgnoreBundle.<clinit>(IgnoreBundle.java:110)
    at mobi.hsz.idea.gitignore.file.IgnoreFileTypeFactory.createFileTypes(IgnoreFileTypeFactory.java:50)
    at com.intellij.openapi.fileTypes.impl.FileTypeManagerImpl.lambda$initStandardFileTypes$0(FileTypeManagerImpl.java:363)
    at com.intellij.openapi.extensions.impl.ExtensionPointImpl.processWithPluginDescriptor(ExtensionPointImpl.java:292)
    at com.intellij.openapi.extensions.ExtensionPointName.processWithPluginDescriptor(ExtensionPointName.java:157)
    at com.intellij.openapi.fileTypes.impl.FileTypeManagerImpl.initStandardFileTypes(FileTypeManagerImpl.java:361)
    at com.intellij.openapi.fileTypes.impl.FileTypeManagerImpl.<init>(FileTypeManagerImpl.java:230)
    at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
    at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
    at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
    at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
    at com.intellij.serviceContainer.ConstructorInjectionKt.instantiateUsingPicoContainer(constructorInjection.kt:44)
    at com.intellij.serviceContainer.PlatformComponentManagerImpl.instantiateClassWithConstructorInjection(PlatformComponentManagerImpl.kt:500)
    at com.intellij.serviceContainer.ServiceComponentAdapter.createAndInitialize(ServiceComponentAdapter.kt:52)
    at com.intellij.serviceContainer.ServiceComponentAdapter.doCreateInstance(ServiceComponentAdapter.kt:39)
    at com.intellij.serviceContainer.BaseComponentAdapter.getInstanceUncached(BaseComponentAdapter.kt:110)
    at com.intellij.serviceContainer.BaseComponentAdapter.getInstance(BaseComponentAdapter.kt:72)
    at com.intellij.serviceContainer.BaseComponentAdapter.getInstance$default(BaseComponentAdapter.kt:65)
    at com.intellij.serviceContainer.PlatformComponentManagerImpl$preloadServices$future$1.run(PlatformComponentManagerImpl.kt:617)
    at java.base/java.util.concurrent.CompletableFuture$AsyncRun.run(CompletableFuture.java:1736)
    at com.intellij.util.concurrency.BoundedTaskExecutor.doRun(BoundedTaskExecutor.java:222)
    at com.intellij.util.concurrency.BoundedTaskExecutor.access$200(BoundedTaskExecutor.java:30)
    at com.intellij.util.concurrency.BoundedTaskExecutor$1.execute(BoundedTaskExecutor.java:201)
    at com.intellij.util.concurrency.BoundedTaskExecutor$1.run(BoundedTaskExecutor.java:193)
    at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
    at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
    at java.base/java.lang.Thread.run(Thread.java:834)
```

![Xnip2019-10-16_17-06-51](https://user-images.githubusercontent.com/17295616/66905340-4223fb80-f038-11e9-8ebc-fbf6b4ea194c.jpg)
![Xnip2019-10-16_17-13-03](https://user-images.githubusercontent.com/17295616/66905421-6aabf580-f038-11e9-95f5-be429b9de6d6.jpg)

**Comment by xinkunZ** [comment]
why you guys haven't release an update for  IDEA 2019.3BETA

**Comment by pl77** [comment]
I'm guess they're still figuring out the cause.  I don't know java very well but it seems JetBrains made some modifications to line 67 of the ConcurrentWeakHashMap.java file around September of this year: 

https://github.com/JetBrains/intellij-community/blob/a49f4faa2b9fc97c4eca661e8352a1fba3391951/platform/util/src/com/intellij/util/containers/ConcurrentWeakHashMap.java#L67

Again, I don't know Java at all but it seems as though line 64 has both keys and values whereas the jetbrains function now only accepts keys and not values.  

https://github.com/JetBrains/idea-gitignore/blob/41c47b81d1ed4ae7f90663dade47fff68322a75f/src/mobi/hsz/idea/gitignore/util/CachedConcurrentMap.java#L64

Maybe that's somewhere to start looking?

**Comment by DavidZemon** [comment]
This problem is getting a bit annoying... been staring at the

> 3:24 PM	Plugin Error: Plugin ".ignore" is incompatible (target build range is 192 to 193).

error every time i start one of my IDEs for way too long now. 2019.3 final should be coming out very soon now...

**Comment by Nothing-Works** [comment]
it's out, now

**Comment by JoshuaLuckers** [comment]
2019.3 seems to have ignore file handling out of the box according to the "[What's New In PhpStorm 2019.3](https://www.jetbrains.com/phpstorm/whatsnew/)" page.

**Comment by dmitriysmirnovjb** [comment]
New v.3.2.3 has been released, should work fine with 2019.3

**Comment by dmitriysmirnovjb** [comment]
> 2019.3 seems to have ignore file handling

Builtin gitignore highlighting has been added in 2019.1, and other gitingore features are builtin since 2019.2. See https://github.com/JetBrains/idea-gitignore/issues/587#issuecomment-514275657
